### PR TITLE
88 avalc2aval

### PR DESCRIPTION
**Pull request by pk505289**
Closes # .

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md`
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
@pk505289 , please change the target branch to `devel`. ("Edit" button at the top)

**Comment by bundfussr** [comment]
@pk505289 , please rename the branch to `88_avalc2aval@devel`. The `@devel` suffix is required to load the devel versions of the upstream dependencies (admiral, admiraldev, and admiral.test). See also https://pharmaverse.github.io/admiraldev/main/articles/git_usage.html#working-with-feature-branches.

File: R/derive_param_response.R [hunk 1]
```diff
@@ -210,13 +219,16 @@ derive_param_response <- function(dataset,
 
   # ---- Select the 1st response and add a new PARAMCD to the input dataset ----
   dataset %>%
-    derive_param_first_event(
+    derive_param_first_eventl(
```
Comment (sgorm123) [inline R/derive_param_response.R:222]: derive_param_first_eventl, is the l at the end expected?

File: R/derive_param_response.R [hunk 2]
```diff
@@ -62,6 +62,14 @@
 #'
 #'   The values must be symbols, character strings, numeric values or `NA`.
 #'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'   analysis value (`AVAL`)
+#'
+#'   The (first) argument of the function must expect a character vector and the
+#'   function must return a numeric vector.
+#'
+#'   *Default:* `yn_to_numeric` (see `admiral::yn_to_numeric()` for details)
```
Comment (sgorm123) [inline R/derive_param_response.R:71]: There was a decision to remove all defaults, so probably better to remove this here too.

File: R/derive_param_response.R [hunk 3]
```diff
@@ -167,6 +175,7 @@ derive_param_response <- function(dataset,
                                   source_pd = NULL,
                                   source_datasets = NULL,
                                   set_values_to,
+                                  aval_fun = yn_to_numeric,
                                   subject_keys = vars(STUDYID, USUBJID)) {
```
[the remark this answers is not in the document]
Comment (sgorm123) [reply R/derive_param_response.R:179]: actually, I think you also need an assertion that AVALC, PARAMCD and ADT are also present, so something like this:

```
assert_data_frame(dataset,
    required_vars = quo_c(subject_keys, vars(PARAMCD, ADT, AVALC))
  )
```

File: R/derive_param_response.R (hunk 2 above)
Comment (bundfussr) [reply R/derive_param_response.R:71]: That's right. However, it provides a link to the default function. Thus I would keep it.

**Comment by bundfussr** [comment]
Yes, "Closes #88" should be added to the title. Once we merge devel into main this will a point of the PR check list.

File: R/derive_param_response.R (hunk 2 above)
Comment (sgorm123) [reply R/derive_param_response.R:71]: I removed all defaults from all other functions in accordance with the response from @rossfarrugia and @thomas-neitmann to this issue: https://github.com/pharmaverse/admiralonco/issues/130

So this is the only function with a default now, in admiralonco.

Comment (rossfarrugia) [reply R/derive_param_response.R:71]: i'd remove it for now for consistency with Stephen's PR we merged that deleted the rest of these. if we get feedback that people are struggling to find this then we could always add a note without using the `*Default:*` convention, but users could always just use `??yn_to_numeric()` if stuck

File: R/derive_param_response.R (hunk 1 above)
Comment (pk505289) [reply R/derive_param_response.R]: This was a typo.. I have updated.

File: R/derive_param_response.R (hunk 2 above)
Comment (pk505289) [reply R/derive_param_response.R]: Removed the *Default.....  from the header

File: R/derive_param_response.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (pk505289) [reply R/derive_param_response.R:178]: added the assertion for Subject Keys and also for the required_vars

**Pull request closed without merging**